### ISSUE TITLE
Issue in passing function pointer as parameter to log functions

### ISSUE THREAD

**Issue by anupama-pathirage** [ballerina-platform/ballerina-standard-library#3738]
**Description:**

Suppose the log level is `INFO`

execute the below code.

```bal
import ballerina/log;
import ballerina/random;

public function main() {
    log:printDebug("Checking the duration", duration = getDutration());
}

function getDutration() returns float {
    log:printInfo("Calculating duration");
    return random:createDecimal() * 100.0;
}
```

It gives output as follows.

```
time = 2022-11-30T01:54:19.061-06:00 level = INFO module = testmodule/testprj message = "Calculating duration"
```

But ideally since `duration = getDutration()` is in a `printDebug`, `getDuration` should not get executed if the log level is INFO.

**Note:**

Following code is working as expected.

```bal
import ballerina/log;
import ballerina/random;

public function main() {
    log:printDebug("Checking the duration", duration = isolated function() returns float {
                       log:printInfo("Calculating duration optimally");
                       return random:createDecimal() * 100.0;});
}
```

**Comment by MadhukaHarith92** [comment]
I checked the issue and it is the expected behavior. In the first example, the `duration` is a `float` value. Therefore, the `getDutration()` function gets invoked before `log:printDebug()` function which is why the `INFO` log inside the `getDutration()` function is getting printed.

In the second example, the `duration` is a function pointer and therefore it doesn't get invoked since the `DEBUG` logs aren't enabled.

I have corrected the section of the article[[1]](https://dzone.com/content/3170704/edit.html) related to this and submitted it for review.

[1] https://dzone.com/content/3170704/edit.html
